test/fixtures/basic/plugins/my-plugin.ts

```diff
@@ -1,12 +1,17 @@
-export default defineNuxtPlugin(() => {
-  useHead({
-    titleTemplate: '%s - Fixture',
-  })
-  const path = useRoute()?.path
-  return {
-    provide: {
-      myPlugin: () => 'Injected by my-plugin',
-      path: () => path,
-    },
-  }
+export default defineNuxtPlugin({
```

Comment:
I don't think `useHead()` was working from plugins for server components previously :thinking: 

We inadvertently fixed this in this PR but then it caused the test snapshots to break, moving this to not run within a server component was the easiest fix.

Reply:
cc: @huang-julien